src/wazuh.py

```diff
@@ -293,22 +349,93 @@ def _get_current_configuration_url_branch(container: ops.Container) -> str:
     return branch.rstrip()
 
 
-def configure_git(
+def pull_config_repo(container: ops.Container, hostname: str, base_url: str, branch: str) -> None:
+    """Pull a local copy of the specified repository.
+
+    Args:
+        container: the container to configure git for.
+        hostname: hostname of the git repository server
+        base_url: base_url of the repository
+        branch: branch or tag name to pull. Can be an empty string.
+
+    Raises:
+        WazuhInstallationError: if an error occurs while configuring git.
+    """
+    logger.info("cloning the '%s' branch of '%s'", branch, base_url)
+    try:
+        process = container.exec(["ssh-keyscan", "-t", "rsa", str(hostname)], timeout=10)
+        output, _ = process.wait_output()
+        container.push(
+            KNOWN_HOSTS_PATH,
+            output,
+            encoding="utf-8",
+            make_dirs=True,
+            user=WAZUH_USER,
+            group=WAZUH_GROUP,
+            permissions=0o600,
+        )
+        _ = container.exec(["rm", "-Rf", REPOSITORY_PATH], timeout=1).wait_output()
+        command = ["git", "clone", "--depth", "1"]
+        if branch:
+            command = command + ["--branch", branch]
+        command = command + [base_url, REPOSITORY_PATH]
+        _ = container.exec(command, timeout=60).wait_output()
+    except ops.pebble.ExecError as ex:
+        logger.error("git clone of custom_config_repository failed")
+        raise WazuhInstallationError from ex
+
+
+def sync_config_repo(
     container: ops.Container,
-    custom_config_repository: typing.Optional[str],
+    custom_config_repository: typing.Optional[AnyUrl],
     custom_config_ssh_key: typing.Optional[str],
-) -> None:
-    """Configure git.
+) -> bool:
+    """Synchronize a local copy of the custom config repository.
 
     Args:
         container: the container to configure git for.
         custom_config_repository: the git repository to add to known hosts in format
-        git+ssh://<user>@<url>:<branch>.
+        git+ssh://<user>@<url>:<branch|tag>.
         custom_config_ssh_key: the SSH key for the git repository.
 
+    Returns:
+        bool: True if a pull was executed, False if there was nothing to sync
+
     Raises:
         WazuhInstallationError: if an error occurs while configuring git.
     """
+    if custom_config_repository is None:
+        # nothing to do
+        return False
+
+    repo_url = str(custom_config_repository)
+
+    base_url = ""
+    branch = ""
+
+    try:
+        url = urlsplit(repo_url)
+        path_parts = url.path.split("@")
+        # 'branch' could be either a branch name or tag name
+        branch = path_parts[1] if len(path_parts) > 1 else ""
+        base_url = urlunsplit(url._replace(path=path_parts[0]))
```

Comment:
would this be a simpler way to do this? i don't think we would have `@` symbol in the scheme or netloc, so the urlsplit would be unnecessary.
```suggestion
        base_url, ref = repo_url.rsplit("@", 1)
```
this also covers an edge case where username has an `@` symbol. I believe the current code would fail in that case.